src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -11259,12 +11259,31 @@ void WaterThermalTankData::SizeTankForSupplySide(EnergyPlusData &state)
     } else if (this->Sizing.DesignMode == SizingMode::PerSolarColArea) {
 
         this->Sizing.TotalSolarCollectorArea = 0.0;
+
         for (int CollectorNum = 1; CollectorNum <= state.dataSolarCollectors->NumOfCollectors; ++CollectorNum) {
-            this->Sizing.TotalSolarCollectorArea += state.dataSurface->Surface(state.dataSolarCollectors->Collector(CollectorNum).Surface).Area;
+            auto const &collector = state.dataSolarCollectors->Collector(CollectorNum);
+            this->Sizing.TotalSolarCollectorArea += state.dataSurface->Surface(collector.Surface).Area;
```

Comment:
* [  ]TODO: Should we make this should only count the collectors that serve this WH?!


```c++
const int thisPlantNum = this->SrcSidePlantLoc.loopNum;
const int plantNum = collector.plantLoc.loopNum;
if (plantNum == thisPlantNum) {
  this->Sizing.TotalSolarCollectorArea += state.dataSurface->Surface(collector.Surface).Area;
}
```

The documentation does say that it uses all of the collectors in the model, but I think that is weird.

https://bigladdersoftware.com/epx/docs/24-1/input-output-reference/group-water-heaters.html#field-design-mode

> PerSolarCollectorArea 
> This design method scales tank volume based on the collector area for a solar hot water collector. **The collector area is summed for all the collectors in the model and each tank is sized for the total**. The collector area is determined from input for Solar Collectors defined elsewhere in the input file.